pipeline-runner/R/qc-7-embed_and_cluster.R

```diff
@@ -14,3 +24,37 @@ embed_and_cluster <- function(scdata, config, sample_id, task_name = 'configureE
 
   return(result)
 }
+
+format_cell_sets_object <- function (cell_sets,type,color_pool) {
+  key <- type
+  name <- paste0(type," clusters")
+
+  #careful with capital l on type for the key.
+  cell_sets_object <- list(key=key,name=name,rootNode=TRUE,type="cellSets",children=list())
+  for(i in sort(unique(cell_sets$cluster))){
+    cells <- cell_sets[cell_sets$cluster==i,"cell_ids"]
+    new_set <- list(
+      key=paste0(key,"-",i),
+      name=paste0("Cluster ",i),
+      rootNode=FALSE,
+      type="cellSets",
+      color=color_pool[1],
+      cellIds=unname(cells)
+    )
+    color_pool <- color_pool[-1]
+    cell_sets_object$children <- append(cell_sets_object$children,list(new_set))
+  }
+  return(cell_sets_object)
+}
+
+update_sets_through_api <- function(cell_sets_object,api_url,experiment_id,cell_set_key){
+  httr::PATCH(
+    paste0(api_url,"/v1/experiments/",experiment_id,"/cellSets"),
+    body = list(list("$match"=list(query=paste0("$[?(@.key == ",cell_set_key,")]"),"$remove"=TRUE)),list("$prepend"=cell_sets_object)),
```

Comment:
this line is difficult to read -- could you format it, so that it is not all in 1 line?